Frontend/ui-driver/src/Screens/DriverProfileScreen/Controller.purs

```diff
@@ -222,6 +288,19 @@ eval (GetDriverInfoResponse (GetDriverInfoResp driverProfileResp)) state = do
                                       driverGender = driverProfileResp.gender
                                       }})
 
+eval (GetRcsDataResponse  (GetAllRcDataResp rcsData)) state = do 
+  let rctransformedData = makeRcsTransformData rcsData 
+      val = rctransformedData
+  if (length val == 1) 
+    then do 
+      let activeRcVal = val 
+      continue state{data{rcsData = val, rcsExceptFirst = drop 1 val, renderingRc = fromMaybe ({ rcStatus : false, rcDetails : {certificateNumber : "", vehicleColor : "", vehicleModel : ""}}) (activeRcVal!!0)}}
+    else do 
+      let activeRcVal = filter (\rc -> rc.rcStatus == true) val
+      if (length activeRcVal == 0) 
+        then continue state {data{rcsData = val, rcsExceptFirst = drop 1 val, renderingRc = fromMaybe ({ rcStatus : false, rcDetails : {certificateNumber : "", vehicleColor : "", vehicleModel : ""}}) (val!!0) }} 
```

Comment:
This `{ rcStatus : false, rcDetails : {certificateNumber : "", vehicleColor : "", vehicleModel : ""}}` can be defined as a dummy value